stdlib.sh

```diff
@@ -654,12 +840,443 @@ function __STDLIB_API_1_info() { # {{{
+#
+# waitForRes "Executing something"
+# evalRes "echo 'hello!'"
+# if (( $STDLIB_EXEC_STATUS == 0 )); then
+#   resOk
+#
+# elif (( $STDLIB_EXEC_STATUS == 128 )); then
+#   resFail "Got the dreaded 128 code!"
+#
+# elif [[ "${STDLIB_EXEC_RESULT}" == "hello!" ]]; then
+#   resOk "It worked! :O"
+#
+# else
+#   resWarning "Something is not right here..."
+# fi
+#
+function __STDLIB_API_1_checkRes() { # {{{
+	local option="${1:-${STDLIB_EXEC_WARNONERR}}"
+	local -i res=${2:-$STDLIB_EXEC_STATUS}
+	local message="${3:-}"
+
+	if [[ "${option}" != "${STDLIB_EXEC_WARNONERR}" ]]  &&  [[ "${option}" != "${STDLIB_EXEC_ERRONERR}" ]]  &&  [[ "${option}" != "${STDLIB_EXEC_FAILONERR}" ]]; then
+		std_ERRNO="$( errsymbol EARGS )"
+		resFail "invalid evaluation option"
+
+	elif (( $res )); then
+		if [[ "${option}" == "${STDLIB_EXEC_WARNONERR}" ]]; then
+			resWarning "${message}"
+
+		elif [[ "${option}" == "${STDLIB_EXEC_ERRONERR}" ]]; then
+			resKo "${message}"
+
+		elif [[ "${option}" == "${STDLIB_EXEC_FAILONERR}" ]]; then
+			resFail "${message}"
+		fi
+	else
+		resOk "${message}"
+	fi
+
+	# std_ERRNO is set in evalRes
+
+	return $res
+} # __STDLIB_API_1_checkRes # }}}
+
+
+###############################################################################
+#
+# stdlib.sh - Standard functions - Map color words to color codes
+#
+###############################################################################
+
+function __STDLIB_API_1_std::mapColorCode() { # {{{
+	local param_colorWord="${1:-}"
+	local -i colorCode=0
+
+	#[[ ! -n "${param_colorWord:-}" ]] && { echo >&2 "Undefined colour"; return 1; }
+	[[ ! -n "${param_colorWord:-}" ]] && return 1
+
+	case $param_colorWord in
+		black)
+			colorCode=0
+			;;
+
+		red)
+			colorCode=1
+			;;
+
+		green)
+			colorCode=2
+			;;
+
+		yellow)
+			colorCode=3
+			;;
+
+		blue)
+			colorCode=4
+			;;
+
+		magenta)
+			colorCode=5
+			;;
+
+		cyan)
+			colorCode=6
+			;;
+
+		white)
+			colorCode=7
+			;;
+
+		*)
+			echo >&2 "Unsupported colour"
+			return 1
+			;;
+	esac
+
+	respond "${colorCode}"
+	return 0
+
+} # __STDLIB_API_1_std::mapColorCode() # }}}
+
+
+###############################################################################
+#
+# stdlib.sh - Standard functions - Map color words to color codes
+#
+###############################################################################
+
+function __STDLIB_API_1_std::mapTextMode() { # {{{
+	local param_modeWord="${1:-}"
+	local modeCode=""
+
+	#[[ ! -n "${param_modeWord:-}" ]] && { echo >&2 "Undefined mode"; return 1; }
+	[[ ! -n "${param_modeWord:-}" ]] && return 1
+
+	case $param_modeWord in
+		bold)
+			modeCode="bold"
+			;;
+
+		dim)
+			modeCode="dim"
+			;;
+
+		underline)
+			modeCode="smul"
+			;;
+
+		reverse)
+			modeCode="rev"
+			;;
+
+		standout)
+			modeCode="smso"
+			;;
+
+		*)
+			echo >&2 "Unsupported mode"
+			return 1
+			;;
+	esac
+
+	respond "${modeCode}"
+	return 0
+
+} # __STDLIB_API_1_std::mapTextMode() # }}}
+
```

Comment:
Are the above two functions useful outside of the context of those below?

If not, then consider a composite `settext <foreground|background|mode> <value>` function?